py/desisim/lya_spectra.py

```diff
@@ -104,10 +114,267 @@ def get_spectra(lyafile, nqso=None, wave=None, templateid=None, normfilter='sdss
         normmaggies = np.array(normfilt.get_ab_maggies(padflux, padwave, 
                                                        mask_invalid=True)[normfilter])
         flux1 *= 10**(-0.4 * mag_g[ii]) / normmaggies
+
+        # Inject a DLA?
+        if add_dlas:
+            if np.min(wave/1215.67 - 1) < zqso[ii]: # Any forest?
+                if debug:
+                    print('ii = {:d}'.format(ii))
+                dla_model = insert_dlas(wave, zqso[ii])
+                flux1 *= dla_model
         flux[ii, :] = flux1[:]
 
     h.close()
 
     return flux, wave, meta
 
 
+def insert_dlas(wave, zem, rstate=None, seed=None, fNHI=None, debug=False, **kwargs):
+    """ Insert zero, one or more DLAs into a given spectrum towards a source
+    with a given redshift
+    Args:
+        wave:
+        zem:
+        rstate:
+        seed:
+        fNHI:
+        **kwargs:
+
+    Returns:
```

Comment:
Please finish this documentation including units, how `**kwargs` is used, and whether `insert_dlas` updates any of the inputs in-place vs. returning a new spectrum with DLAs while leaving the original untouched.  Thanks for carefully documenting the docstrings of the other functions added here.